test/testcafe/spec/ChallengeOktaVerifyFastPassView_spec.js

```diff
@@ -51,6 +55,80 @@ const loopbackSuccesskMock = RequestMock()
     'access-control-allow-methods': 'POST, OPTIONS'
   });
 
+const loopbackPollingLogger = RequestLogger(/introspect|probe|challenge|poll/, { logRequestBody: true, stringifyRequestBody: true });
+const loopbackRedundantPollingMock = RequestMock()
+  .onRequestTo(/\/idp\/idx\/introspect/)
+  .respond(identifyWithUserVerificationLoopback)
+  .onRequestTo(/\/idp\/idx\/authenticators\/poll/)
+  .respond((req, res) => {
+    res.headers['content-type'] = 'application/json';
+    if (!firstPollCalledForRedundantPolling) {
+      firstPollCalledForRedundantPolling = true;
+      // specifically make the first poll call takes more time to complete
+      // so that we can verify if there will be a redundant poll call after it
+      return new Promise((resolve) => setTimeout(function() {
+        res.statusCode = '200';
+        res.setBody(identify);
+        resolve(res);
+      }, 8000));
```

Comment:
instead of returning a new promise, is it possible to instead do this (only suggesting this to possibly avoid flakiness):
```suggestion
      await new Promise((r) => setTimeout(r, 8000));
      res.statusCode = '200';
      res.setBody(identify);
```

Reply:
Also, just curious, for the sake of this test, is the extended delay necessary? We try to avoid this sort of delay unless it to to cover a scenario that cant be tested otherwise.

Reply:
yes it needs to assert three views:
1. device challenge polling view
2. bad request error view
3. identify page view(mocking a successful redirect)
2 and 3 are both response from polling. and we need to make sure that 3 happens first but responses are returning after 2. That is the issue that we fixed before.